### PR TITLE
Rewrite hdf

### PR DESCRIPTION
**Pull request by samwaseda**

### CODE REVIEW

File: pyiron/base/job/interactive.py [hunk 1]
```diff
@@ -263,6 +265,66 @@ def interactive_open(self):
...
+        return hdf_new
+
+    def file_size(self, hdf):
+        """
+        args:
+            hdf (ProjectHDFio): hdf file
+        """
+        return os.path.getsize(hdf.file_name)
+
+    def get_size(self, hdf):
```
Comment (jan-janssen) [inline pyiron/base/job/interactive.py]: Can not this be implemented in pyiron.base.generic.hdfio.ProjectHDFio?

File: pyiron/base/job/interactive.py [hunk 2]
```diff
@@ -263,6 +265,66 @@ def interactive_open(self):
...
+        for p in hdf_old.list_nodes():
+            hdf_new[p] = hdf_old[p]
+        for p in hdf_old.list_groups():
+            if p in exclude_groups:
+                continue
+            h_new = hdf_new.create_group(p)
+            self.hd_copy(hdf_old[p], h_new, exclude_groups=exclude_groups)
+        return hdf_new
+
+    def file_size(self, hdf):
```
Comment (jan-janssen) [inline pyiron/base/job/interactive.py]: Can not this be implemented in pyiron.base.generic.hdfio.ProjectHDFio ?

File: pyiron/base/job/interactive.py [hunk 3]
```diff
@@ -263,6 +265,66 @@ def interactive_open(self):
         """
         self.server.run_mode.interactive = True
 
+    def hd_copy(self, hdf_old, hdf_new, exclude_groups = []):
```
Comment (jan-janssen) [inline pyiron/base/job/interactive.py]: Can not this be implemented in pyiron.base.generic.hdfio.ProjectHDFio ?

File: pyiron/base/job/interactive.py [hunk 4]
```diff
@@ -263,6 +265,66 @@ def interactive_open(self):
...
+            hdf (ProjectHDFio): hdf file
+        """
+        m = 0
+        for p in hdf.list_nodes():
+            m += sys.getsizeof(hdf[p])
+        for p in hdf.list_groups():
+            m += self.get_size(hdf[p])
+        return m
+
+    def rewrite_hdf5(self, info=False, exclude_groups=['interactive']):
```
Comment (jan-janssen) [inline pyiron/base/job/interactive.py]: The interactive group might be required if we restart the execution of the job in a separate run. In addition I could see this function being implemented in pyiron.base.job.core.JobCore

File: pyiron/base/job/interactive.py (hunk 1 above)
Comment (samwaseda) [reply pyiron/base/job/interactive.py]: ok I'll try

**Comment by samwaseda** [comment]
ok I saw [this file](https://github.com/pyiron/pyiron/blob/master/tests/static/dft/es_hdf.h5) but this doesn't exist?

**Pull request merged**